### PR TITLE
Address resolver customer id refactor

### PR DESCRIPTION
**Pull request by ahmardi**

### CODE REVIEW

File: Model/AddressResolver.php [hunk 1]
```diff
@@ -76,59 +76,56 @@ public function __construct(
...
-                $amwalAddressId
-            )->addFilter(
-                'parent_id',
-                $this->getCustomerId()
-            )->create();
-            $matchedAddresses = $this->addressRepository->getList($searchCriteria)->getItems();
-            if ($matchedAddresses) {
-                $address = reset($matchedAddresses);
-                $this->updateTmpAddressData($address, $amwalOrderData);
+        if(!empty($customerId)){
```
Comment (sameh-amwal) [inline Model/AddressResolver.php:87]:
```suggestion
        if($customerId){
```

File: Model/AddressResolver.php [hunk 2]
```diff
@@ -76,59 +76,56 @@ public function __construct(
...
-                $amwalAddressId
-            )->addFilter(
-                'parent_id',
-                $this->getCustomerId()
-            )->create();
-            $matchedAddresses = $this->addressRepository->getList($searchCriteria)->getItems();
-            if ($matchedAddresses) {
-                $address = reset($matchedAddresses);
-                $this->updateTmpAddressData($address, $amwalOrderData);
+        if($customerId){
```
Comment (Skullsneeze) [inline Model/AddressResolver.php:87]:
```suggestion
        if ($customerId) {
```

**Review by Skullsneeze — APPROVED** [review]

File: Model/Checkout/GetQuote.php [hunk 3]
```diff
@@ -288,7 +288,7 @@ public function createQuote(array $orderItems): Quote
         $quote->setStore($this->storeManager->getStore());
         $quote->setCurrency();
 
-        if ($customer = $this->getCustomer()) {
+        if ($customer = $this->getSessionCustomer()) {
             $quote->assignCustomer($customer);
         } else {
```
Comment (sameh-amwal) [inline Model/Checkout/GetQuote.php:293]: Only setCustomerIsGuest if the quote doesn't have customer id. 
```suggestion
        } else if (!quote->getCustomerId()) {
```

File: Test/Unit/Model/Checkout/GetQuoteTest.php [hunk 4]
```diff
@@ -83,6 +83,7 @@ class GetQuoteTest extends TestCase
     private const AMWAL_ADDRESS_ID = '6e369835-451c-4071-8d86-496bd4a19eb6';
     private const TRIGGER_CONTEXT = 'cart';
     private const ORDER_ITEMS = [];
+    private const CUSTOMER_ID = null;
```
Comment (sameh-amwal) [inline Test/Unit/Model/Checkout/GetQuoteTest.php:86]: you should delete `IS_GUEST_QUOTE` variable

**Review by sameh-amwal — APPROVED** [review]

**Pull request merged**